pkg/storage/ingest/DESIGN.md

```diff
@@ -0,0 +1,18 @@
+# Ingest storage design
+
+This document aims to summarise some design principles and technical decisions applied when building the ingest storage.
+
+## Kafka offsets
+
+### Last produced offset
+
+The offset of the **next** record in a partition can be read from Kafka issuing `ListOffsets` request with `timestamp = -1`.
+The special value `-1` means "latest" but in practice it's the "last produced offset + 1".
+
+This has been verified both testing it in Confluent Kafka 7.5 and Warpstream v524. In details:
```

Comment:
I meant Confluent Kafka because that's what we run in our local dev env (see: https://github.com/grafana/mimir/compare/write-to-kafka#diff-0f660ca834bff4dfe25e623a78d4fedf70dc3a165120c93b7a0df66ea67c1010R108) but Confluent Kafka 7.5 should be the packaging of Apache Kafka 3.5 ([release notes](https://docs.confluent.io/platform/current/release-notes/index.html)), so let me just change it to make it clearer.